### ISSUE TITLE
Feature: "remap" injected interface methods

### ISSUE THREAD

**Issue by TropheusJ** [FabricMC/fabric-loom#859]
Injected interfaces are very useful, but they are dangerous. They might conflict with other mods injecting interfaces into the same target. This can be fixed by prefixing methods with a mod ID, but that makes the added methods messy to use.

Ideally, prefixes would not be present in dev, but would be in production, getting all the benefits. This would be essentially the same as remapping the methods, right? For a dev name of `doThing`, have an "intermediary" name of `mod_id$doThing`, just prefixing the mod ID. This renaming would be done by loom when building the mod.

When a mod adds a mod with injected interfaces to its dev environment, the names will be mapped back to their dev names, just removing the prefix. In case of a collision in a dev environment, there should also be a way to disable this behavior for specific methods.

**Comment by modmuss50** [comment]
I dont think we can easily do this, you should try to prefix your method names. You would do the same even when using interfaces with mixin.
